OWML.ModHelper/ModConfig.cs

```diff
@@ -65,5 +65,57 @@ public T GetSetting<T>(string key)
             return GetSettingsValue<T>(key);
         }
 
+        private void AddMissingDefaults(IModConfig defaultConfig)
+        {
+            foreach (var defaultSetting in defaultConfig.Settings)
+            {
+                if (!Settings.ContainsKey(defaultSetting.Key))
+                {
+                    Settings.Add(defaultSetting.Key, defaultSetting.Value);
+                }
+            }
+        }
+
+        public void MakeConfigConsistentWithDefaults(IModConfig defaultConfig)
+        {
+            if (defaultConfig == null)
+            {
+                return;
+            }
+
+            AddMissingDefaults(defaultConfig);
+
+            var settingsCopy = new Dictionary<string, object>(Settings);
+            foreach (var setting in Settings)
+            {
+                if (!defaultConfig.Settings.ContainsKey(setting.Key))
+                {
+                    settingsCopy.Remove(setting.Key);
+                    continue;
+                }
+                else if (!IsSettingSameType(setting.Value, defaultConfig.Settings[setting.Key]))
```

Comment:
maybe add separate detection if type is similar to not override the value? (i.e. number -> slider)

Reply:
or string -> combination

Reply:
Sounds like something for another PR. Don't see the need to complicate this further.

Reply:
Sounds like it might require some manual coding though

Reply:
Have you not been coding manual until now? 🎷

Reply:
I mean it doesn't sound like it could be solved with single `if`